frame/assets/src/functions.rs

```diff
@@ -94,11 +92,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			},
 			ExistenceReason::DepositRefunded => {},
 			ExistenceReason::DepositHeld(_) if !force => return Keep,
-			ExistenceReason::DepositHeld(_) => result = Keep,
+			ExistenceReason::DepositHeld(_) => {},
```

Comment:
For other reviewer, this change makes sense because in case we pass `force` then we force the removal of the account, we also substract 1 to the account counter of the asset. So the account must be removed by the caller.

In practice the current code only use force in `destroy_asset` and correctly remove the account.